lightning/src/chain/keysinterface.rs

```diff
@@ -311,6 +311,17 @@ pub trait ChannelKeys : Send+Clone {
 	/// chosen to forgo their output as dust.
 	fn sign_closing_transaction<T: secp256k1::Signing>(&self, closing_tx: &Transaction, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()>;
 
+	/// Create a signature for a Child-Pay-For-Parent transaction.
+	///
+	/// Such a transaction is used to unilaterally bump a commitment transaction feerate when this
+	/// is one isn't confirming quickly.
+	///
+	/// Input index is the position of the spending input committed by this BIP 143 signature.
+	///
+	/// Spent amount is the value of the output spent by this input committed by this BIP 143
+	/// signature.
+	fn sign_cpfp<T: secp256k1::Signing>(&self, cpfp: &Transaction, input_index: usize, spent_amount: u64, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()>;
```

Comment:
any chance we can break up `cpfp` into components (in the spirit of the phase 2 signer API), or is it premature because we don't know yet what are all the shapes we might want for the transaction?